kratos/utilities/geometrical_projection_utilities.h

```diff
@@ -199,6 +198,35 @@ class GeometricalProjectionUtilities
         return norm_2(rPointProjected.Coordinates()-p_c);
     }
 
+
+    /**
+     * @brief Computes the minimal distance to a line
+     * @details Projects over a line and if the point porjected is inside the line that distance is taken into consideration, otherwise the minimal between the two points in the line is considered
```

Comment:
Isnt this the same a the new project methods that @tteschemacher added?

Also there is a typo

Reply:
If it is already implemented perfect, can you tell me where?

Reply:
I don't think I have implemented the exact same function. However, you can actually check to include this in the respectively derived function under:
https://github.com/KratosMultiphysics/Kratos/blob/8a9721a05c73c8976139705dc1880d50389a6e74/kratos/geometries/geometry.h#L2660
or 
https://github.com/KratosMultiphysics/Kratos/blob/8a9721a05c73c8976139705dc1880d50389a6e74/kratos/geometries/geometry.h#L2687

Reply:
wait. sorry I was confused.

This one should be the correct interface, no?:
https://github.com/KratosMultiphysics/Kratos/blob/8a9721a05c73c8976139705dc1880d50389a6e74/kratos/geometries/geometry.h#L2881

Reply:
No, it is not the same, here I am computing the closest distance between a line and a point

Reply:
> wait. sorry I was confused.
> 
> This one should be the correct interface, no?:
> 
> https://github.com/KratosMultiphysics/Kratos/blob/8a9721a05c73c8976139705dc1880d50389a6e74/kratos/geometries/geometry.h#L2881

But this one, then?

Reply:
I didn't know about that, I will see what can I refactor

Reply:
I cannot derive this method because the definition is:

~~~c++
    /**
    * @brief Computes the distance between an point in
    *        global coordinates and the closest point
    *        of this geometry.
    *        If projection fails, double::max will be returned.
    * @param rPointGlobalCoordinates the point to which the
    *        closest point has to be found.
    * @param Tolerance accepted orthogonal error.
    * @return Distance to geometry.
    *         positive -> outside of to the geometry (for 2D and solids)
    *         0        -> on/ in the geometry.
    */
~~~

In my case I project, but if it fails I return the closest distance between the nodes of the line. Not exactly the same, if you want to create new method like `CalculateProjectedDistance` or `ClaculateClosestDistance`, etc..., Ok. But this method for me doesn't meet requirements